docs/USER_RECORD.md

```diff
@@ -543,6 +554,15 @@ it is bypassed.
 auto-login. Systems are supposed to automatically log in a user marked this way
 during boot, if there's exactly one user on it defined this way.
 
+`preferredSessionType` → A string that indicates the user's preferred session type
+(i.e. `x11`, `wayland`, or other values valid for `$XDG_SESSION_TYPE`). This should
+be used by the display manager to pre-select the correct environment to log into.
+
+`preferredSessionName` → A string that indicates the user's preferred  (i.e. `gnome`,
```

Comment:
preferredSessionDesktop? or preferredDesktop? as per earlier comments, and that we should keep this close to:

https://wiki.archlinux.org/title/Environment_variables#Examples

Reply:
I'd just call it "preferredDesktop" btw, drop the word "session"

Reply:
Well that's a tad annoying. Github wiped my comment out of existence :eyes: I suppose I'll repeat its gist

`preferredSessionName` is not the same thing as XDG_SESSION_DESKTOP OR XDG_CURRENT_DESKTOP. They're different identifiers with different syntax and different purpose

`preferredSessionName` refers to a file name to look for in this pattern: `/usr/share/{wayland-,x}sessions/*.desktop`. GDM uses this to decide which .desktop file to read, which will give it the command to run to start the session.

For example: if `preferredSessionName` is `gnome`, then GDM will pick `/usr/share/wayland-sessions/gnome.desktop`. If `preferredSessionName` is `gnome-x11`, then GDM will pick `/usr/share/xsessions/gnome-x11.desktop`. If `preferredSessionName` is `gnome` but GDM has Wayland support turned off, it'll look at `/usr/share/xsessions/gnome.desktop`.

XDG_{CURRENT,SESSION}_DESKTOP are for apps to determine what DE they're running on. `sessionName` is for the display manager to decide which command it needs to run to get the session started

Reply:
urks, why are they no using the same identifier?

This feels awfully gdm specific, doesn't it. can't we generalize this somehow?

i mean, with the combination the two fields that use the XDG_SESSION_DESKTOP + XDG_SESSION_TYPE values, gdm could just deduce what it needs, no?

i.e. i.e. XDG_SESSION_TYPE tells it which dir to preferrably look in, and XDG_SESSION_DESKTOP tells it which file to pick from it?

Reply:
I am almost certain that this is not GDM specific behavior. Pretty sure this is the de-facto standard behavior for display managers. I too would re-design this whole thing a lot more reasonably but this what we have. Pretty sure all this predates XDG_SESSION_*, and _probably_ predates systemd and Wayland.

Reply:
Let's continue the discussion in #31310